### PR TITLE
MERGE symoblic-pr to master: STEP one

### PR DESCRIPTION
**Pull request by yzhang90**
symbolic: Created module symbolic.md, Symbolic rules for ECREC.

edsl, symbolic: added generalized byteStack2IntList(), refactoring of module dependencies. Now EVM-SYMBOLIC imports EDSL.

### CODE REVIEW

File: evm.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ module EVM
     imports STRING
     imports EVM-DATA
     imports NETWORK
+    imports K-REFLECTION
```
Comment (daejunpark) [inline evm.md]: i think this can be dropped now.

File: evm-symbolic.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    //Symbolic wrapper over the argument of #ecrec, no implementation.
+    syntax Int ::= #symEcrec ( WordStack )   [function]
+
+    //Symbolic predicate representing whether output of #ecrec is empty. No implementation.
+    syntax Bool ::= #ecrecEmpty( WordStack ) [function]
+```
+
+```k
+endmodule
+```
```
Comment (daejunpark) [inline evm-symbolic.md]: if we do not have much thing to add in this module, i think we can just move these two syntax declarations into EDSL.

Comment (yzhang90) [reply evm-symbolic.md]: I added a few other things in the later commits. Lets keep it for now

File: evm.md (hunk 1 above)
Comment (yzhang90) [reply evm.md]: fixed

Comment (yzhang90) [reply evm.md]: fixed

File: edsl.md [hunk 3]
```diff
@@ -312,5 +312,22 @@ Specifically, `#hashedLocation` is defined as follows, capturing the storage lay
     rule intList2ByteStack(.IntList) => .WordStack
     rule intList2ByteStack(V VS)     => #padToWidth(32, #asByteStack(V)) ++ intList2ByteStack(VS)
       requires 0 <=Int V andBool V <Int pow256
+      
+    syntax IntList ::= byteStack2IntList ( WordStack )       [function]
+                     | byteStack2IntList ( WordStack , Int ) [function]
+                     
+    syntax Int ::=     bytesInNextInt ( WordStack , Int )    [function] //how many bytes will go into next Int.
+ // ------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (yzhang90) [reply edsl.md]: fixed

**Comment by ehildenb** [comment]
Jenkins: test mantis

File: evm-symbolic.md (hunk 2 above)
Comment (daejunpark) [reply evm-symbolic.md]: @ehildenb https://github.com/runtimeverification/verified-smart-contracts/pull/148/files

**Comment by daejunpark** [comment]
byteStack2IntList is used in:
https://github.com/runtimeverification/verified-smart-contracts/blob/master/resources/lemmas.md

**Comment by yzhang90** [comment]
Jenkins: test this please

**Pull request merged**